testsuite/features/secondary/srv_rename_hostname.feature

```diff
@@ -21,6 +21,49 @@ Feature: Reconfigure the server's hostname
     When I change the server's short hostname from hosts and hostname files
     And I run spacewalk-hostname-rename command on the server
 
+@proxy
+  Scenario: Copy the new server keys and configure the proxy
+    When I copy server's keys to the proxy
+    And I configure the proxy
+    Then I should see "proxy" via spacecmd
+    And service "salt-broker" is active on "proxy"
+
+@sle_minion
+  Scenario: Apply high state on the SUSE Minion to populate new server CA
+    When I apply highstate on "sle_minion"
+
+@ssh_minion
+  Scenario: Apply high state on the SUSE SSH Minion to populate new server CA
+    When I apply highstate on "ssh_minion"
+
+@rhlike_minion
+  Scenario: Apply high state on the Red Hat-like Minion to populate new server CA
+    When I apply highstate on "rhlike_minion"
+
+@deblike_minion
+  Scenario: Apply high state on the Debian-like Minion to populate new server CA
+    When I apply highstate on "deblike_minion"
+
+@buildhost
+  Scenario: Apply high state on the build host to populate new server CA
+    When I apply highstate on "build_host"
+
+@virthost_kvm
+  Scenario: Apply high state on the virthost to populate new server CA
+    When I apply highstate on "kvm_server"
+
+@pxeboot_minion
+  Scenario: Apply high state on the PXE boot minion to populate new server CA
+    When I apply highstate on "pxeboot_minion"
```

Comment:
I would use SSM to apply high state to all of them at the same time.

Reply:
Good idea. I still have issues with high state for the proxy when doing it with Salt:
```bash
  @proxy
  Scenario: Apply high state on the proxy to populate new server CA # features/secondary/srv_rename_hostname.feature:67
      This scenario ran at: 2023-10-18 09:37:30 +0200
    Given the Salt master can reach "proxy"                         # features/step_definitions/salt_steps.rb:11
      It took 1 seconds to contact the minion
    And I apply highstate on "proxy"                                # features/step_definitions/command_steps.rb:241
      Timeout after 250 seconds (repeat_until_timeout), last result was: bash: dominik-pxy.mgr.suse.de: command not found
       (RuntimeError)
      ./features/support/commonlib.rb:117:in `block in repeat_until_timeout'
      ./features/support/commonlib.rb:100:in `repeat_until_timeout'
      ./features/support/lavanda.rb:193:in `run_until_ok'
      ./features/step_definitions/command_steps.rb:248:in `/^I apply highstate on "([^"]*)"$/'
      features/secondary/srv_rename_hostname.feature:69:in `I apply highstate on "proxy"'
```
 which uses

```ruby
When(/^I apply highstate on "([^"]*)"$/) do |host|
  system_name = get_system_name(host)
  if host.include? 'ssh_minion'
    cmd = 'mgr-salt-ssh'
  elsif host.include? 'minion' or host.include? 'build'
    cmd = 'salt'
  end
  get_target('server').run_until_ok("#{cmd} #{system_name} state.highstate")
end
```
And running `salt dominik-pxy.mgr.suse.de state.highstate`manually from the server works.

Reply:
After wondering about SSM, I think we do not yet have a step to select all systems from the systems list and add them to the SSM.

Reply:
Here a new proposal scenario that will apply high-state in all registered systems:
```
  Scenario: Apply high-state to all systems registered
    When I follow the left menu "System > System List > All"
    And I click on "Select All"
    And I follow the left menu "Systems > System Set Manager > Overview"
    And I click on "Apply"
    And I click on "Apply Highstate"
    And I click on the clear SSM button
```

Reply:
Thanks for the suggestion. I will do some test run with this code.

Reply:
```bash
  Scenario: Apply high-state to all systems registered                   # features/secondary/srv_rename_hostname.feature:34
      This scenario ran at: 2023-10-20 16:35:03 +0200
    When I follow the left menu "System > System List > All"             # features/step_definitions/navigation_steps.rb:363
      WARN: Step ends with an ajax transition not finished, let's wait a bit!
    And I click on "Select All"                                          # features/step_definitions/navigation_steps.rb:282
    And I follow the left menu "Systems > System Set Manager"            # features/step_definitions/navigation_steps.rb:363
      WARN: Step ends with an ajax transition not finished, let's wait a bit!
    And I click on "Apply"                                               # features/step_definitions/navigation_steps.rb:282
      Unable to find button "Apply" that is not disabled (Capybara::ElementNotFound)
      ./features/support/commonlib.rb:146:in `click_button_and_wait'
      ./features/step_definitions/navigation_steps.rb:283:in `/^I click on "([^"]*)"$/'
      features/secondary/srv_rename_hostname.feature:38:in `I click on "Apply"'
    And I click on "Apply Highstate"                                     # features/step_definitions/navigation_steps.rb:282
    Then I should see a "Applying the highstate has been scheduled" text # features/step_definitions/navigation_steps.rb:599
    And I click on the clear SSM button                                  # features/step_definitions/navigation_steps.rb:799
=> /var/log/rhn/rhn_web_ui.log
```
![image](https://github.com/uyuni-project/uyuni/assets/12104291/355a5f5c-8ccc-45b3-a368-2059ce8ab3aa)